buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchCluster.java

```diff
@@ -293,21 +295,25 @@ public ElasticsearchNode singleNode() {
         return getFirstNode();
     }
 
-    private void addWaitForUri(String description, String uri) {
-        waitConditions.put(description, (node) -> {
+    private void addWaitForClusterHealth() {
+        waitConditions.put("cluster health yellow", (node) -> {
             try {
-                URL url = new URL("http://" + getFirstNode().getHttpSocketURI() + uri);
-                HttpURLConnection con = (HttpURLConnection) url.openConnection();
-                con.setRequestMethod("GET");
-                con.setConnectTimeout(500);
-                con.setReadTimeout(500);
-                try (BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
-                    String response = reader.lines().collect(Collectors.joining("\n"));
-                    LOGGER.info("{} -> {} ->\n{}", this, uri, response);
+                WaitForHttpResource wait = new WaitForHttpResource(
+                    "http", getFirstNode().getHttpSocketURI(), nodes.size()
+                );
+                List<Map<String, String>> credentials = getFirstNode().getCredentials();
+                if (getFirstNode().getCredentials().isEmpty() == false) {
+                    wait.setUsername(credentials.get(0).get("useradd"));
+                    wait.setPassword(credentials.get(0).get("-p"));
                 }
-                return true;
+                return wait.wait(500);
             } catch (IOException e) {
                 throw new IllegalStateException("Connection attempt to " + this + " failed", e);
+            } catch (InterruptedException e) {
```

Comment:
The reason I didn't propagate is that we retry on all exceptions when dealing with wait conditions, so would end up eating it there, but we do have a mechanism to prevent that, I'm pushing a fix, thanks for noticing.